.github/workflows/build.yml

```diff
@@ -1,105 +1,347 @@
+        portable: [Non-Portable]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Create Build Environment
+        run: |
+          if [ ${{ matrix.arch }} == "x86" ]; then
+            sudo dpkg --add-architecture i386
+            sudo apt-get -qq update
+            sudo apt-get -y install aptitude
+            sudo apt-get -y install gcc-multilib g++-multilib ninja-build
+            sudo apt-get -y install --allow-downgrades libpcre2-8-0:i386 libjpeg-dev:i386 libpng-dev:i386 libcurl4-openssl-dev:i386
+            sudo aptitude -y install libglib2.0-dev:i386 libsdl2-dev:i386
+          else
+            sudo apt-get -qq update
+            sudo apt-get install libjpeg-dev libpng-dev zlib1g-dev libsdl2-dev
+          fi
+          cmake -E make_directory ${{ runner.workspace }}/build
+
+      - name: Configure CMake
+        shell: bash
+        working-directory: ${{ runner.workspace }}/build
+        run: |
+          if [ "${{ matrix.portable }}" == "Portable" ]; then
+            OPTIONS="-DUseInternalLibs=ON -DBuildPortableVersion=ON -DCMAKE_INSTALL_PREFIX=${{ runner.workspace }}/install"
+          else
+            OPTIONS="-DUseInternalLibs=OFF -DBuildPortableVersion=OFF -DCMAKE_INSTALL_PREFIX=${{ runner.workspace }}/install"
+          fi
+          
+          if [ ${{ matrix.arch }} == "x86" ]; then
+          cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} $OPTIONS -DCMAKE_TOOLCHAIN_FILE=cmake/Toolchains/linux-i686.cmake
+          else
+          cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} $OPTIONS
+          fi
+
+      - name: Build
+        working-directory: ${{ runner.workspace }}/build
+        shell: bash
+        run: cmake --build .
+
+      - name: Install
+        if: ${{ matrix.build_type == 'Release' }} 
+        working-directory: ${{ runner.workspace }}/build
+        shell: bash
+        run: cmake --install .
+
+      - name: Create binary archive
+        if: ${{ matrix.build_type  == 'Release'}}
+        working-directory: ${{ runner.workspace }}/install/JediAcademy
+        shell: bash
+        run: |
+          if [ ${{ matrix.arch }} == "x86" ]; then
+          chmod +x openjk.i386
+          else
+          chmod +x openjk.${{ matrix.arch }}  
+          fi
+          tar -czvf OpenJK-linux-${{ matrix.arch }}-${{ matrix.build_type }}-${{ matrix.portable }}.tar.gz      *
+
+      - uses: actions/upload-artifact@v3
+        if: ${{ matrix.build_type == 'Release' }} 
+        with:
+          name: OpenJK-linux-${{ matrix.arch }}-${{ matrix.build_type }}-${{ matrix.portable }}
+          path: ${{runner.workspace}}/install/JediAcademy/OpenJK-linux-${{ matrix.arch }}-${{ matrix.build_type }}-${{ matrix.portable }}.tar.gz
+          if-no-files-found: error
 
   macos:
-    name: Build macOS AppleClang
-    runs-on: macos-10.15
+    name: macOS ${{ matrix.arch }} ${{ matrix.build_type }} (${{ matrix.portable}})
+    runs-on: macos-12
```

Comment:
I think we used an older version of macOS for building before so that the mac builds can run on older macOS versions. Does building this on macOS 12 mean we lose support for those older OSs?

Reply:
Older Mac support out in the wild is pretty scarce now.

Reply:
The builds weren't uploaded before either way

Reply:
Good point!